### ISSUE TITLE
Undocumented GDAL 2.3.0 CSV output geometry column name change for GML source

### ISSUE THREAD

**Issue by bencaradocdavies** [OSGeo/gdal#660]
## Expected behaviour and actual behaviour.

CSV output is [documented](http://www.gdal.org/drv_csv.html) to write a geometry column with a namethat defaults to `WKT`. From 2.3.0, CSV output created from a GML file with a geometry property with local name `geometry` instead has a geometry column named `geometry`.

Misbehaving on Debian sid:

```
$ ogr2ogr --version
GDAL 2.3.0, released 2018/05/04
$ ogr2ogr -f CSV -lco GEOMETRY=AS_WKT feature.csv feature.gml; grep STATE feature.csv 
geometry,gml_id,STATE_NAME
```

Behaving as documented on Ubuntu Bionic:

```
$ ogr2ogr --version
GDAL 2.2.3, released 2017/11/20
$ ogr2ogr -f CSV -lco GEOMETRY=AS_WKT feature.csv feature.gml; grep STATE feature.csv
WKT,gml_id,STATE_NAME
```

I am pretty sure that 2.2.4 worked the same as 2.2.3 (as documented).

## Steps to reproduce the problem.

```
ogr2ogr -f CSV -lco GEOMETRY=AS_WKT feature.csv feature.gml; grep STATE feature.csv
```

See: [feature.gml](https://osgeo-org.atlassian.net/secure/attachment/33029/feature.gml) attached to [GEOS-8765](https://osgeo-org.atlassian.net/browse/GEOS-8765.).

## Operating system

Debian sid amd64

## GDAL version and provenance

gdal-bin 2.3.0+dfsg-1 amd64

**Comment by bencaradocdavies** [comment]
Thanks @rouault. Much appreciated.

**Comment by rouault** [comment]
The CSV driver hadn't changed in this regard, but ogr2ogr generic mechanics indirectly triggered this change of behaviour

**Comment by bencaradocdavies** [comment]
Yes, I suspected as much when I found no suspect changes in the CSV driver source. I think the new behaviour was nicer, but sadly it broke  backwards compatibility and and was inconsistent with the documentation. No good deed goes unpunished.
